pkg/storegateway/batch_series.go

```diff
@@ -324,10 +327,13 @@ func (s *BucketStore) batchSetsForBlocks(ctx context.Context, req *storepb.Serie
 	begin := time.Now()
 	err := g.Wait()
 	if err != nil {
-		return nil, nil, nil, cleanup, err
+		return nil, nil, cleanup, err
 	}
-	stats.blocksQueried = len(batches)
-	stats.getAllDuration = time.Since(begin)
+
+	stats.update(func(stats *queryStats) {
+		stats.blocksQueried = len(batches)
+		stats.getAllDuration = time.Since(begin)
+	})
 
```

Comment:
Her we are also missing adding these to the prom metrics 
```suggestion

```
```suggestion
  getAllDuration := time.Since(begin)
	stats.update(func(stats *queryStats) {
		stats.blocksQueried = len(batches)
		stats.getAllDuration = getAllDuration
	})

	s.metrics.seriesGetAllDuration.Observe(getAllDuration.Seconds())
	s.metrics.seriesBlocksQueried.Observe(float64(len(batches)))
```

Reply:
Good point, done.